### DISCUSSION TITLE
[Protokoll] 07.05.2024

### DISCUSSION THREAD

**Discussion by gabrieltorresgamez** [AdversarialAttacks/.github#31] [Protokolle]
### Wer hat an dieser Sitzung teilgenommen?

- [X] Gabriel Torres Gamez
- [X] Si Ben Tran
- [X] Daniel Perruchoud
- [X] Stephan Heule

### Welche sind die wichtigsten Themen, die an dieser Sitzung besprochen wurden?

Kurze Zusammenfassung über den Stand der UAPs für Stephan und Besprechung des Aufbaus und Inhalt des Berichts.

### Was wurde an der Sitzung beschlossen?

Wir müssen uns folgende Gedanken oder Änderungen am Bericht machen:
- Krebsklassenanalyse: Welche Krebsarten haben wir und kann man da zwischen Bösartig und nicht Bösartig unterscheiden.
- Orthographiecheck: Wir schreiben die gleichen Sachen recht unterschiedlich (z.B. BinaryCrossEntropy | Binary Cross Entropy | Binary-Cross-Entropy. Hier müssen wir einheitlicher schreiben! Eventuell ein Orthographietool für LaTeX suchen.
- Uns fehlt die Datenexplorierung (Nicht stark auf Testdaten schauen!).
- Wir sollten unsere Technische Umgebung sowie genutzte Libraries etc. dokumentieren.
- Wir sollten eine schematische Skizze erstellen, welche unser ganzes Projekt beschreibt. Diese sollten wir am besten vor dem Preprocessing erwähnen.
- Algorithmen, Preprocessing etc. sollte mehr theoretisch als praktisch dokumentiert sein (Library-unabhängig).
- Baselinetraining Ergebnisse sollten im Kapitel 5 sein.
- Baseline Modelle sollten wir umbennen, da der Begriff "Baseline" in anderen Kontexten genutzt wird.
- Bei der Fooling Rate sollten wir für den booleanischen Begriff die Indikatorfunktion (\mathcal) nutzen.
- Die Roundfunktion muss nicht mathematisch definiert werden, sie kann im Prosa beschrieben werden.
- Speziell betonen, dass wir die Loss Funktion so optimiert haben um das Optimierungsproblem vom Paper zu lösen
- Bei der UAP Loss Function beschreiben, welche Parameter genau optimiert werden (delta_v).
- Bei der Loss funktion beschreiben, wie sich yhat_adv Zusammensetzt (img + v + delta_v).

Ideen:
- Leistungsaufwand für die Findung der Perturbation auf geschützte Modelle können wir auch betrachten, da es sicher interessant ist (z.B. “Es braucht jetzt 5x länger, um eine Fooling Rate von 50% zu kommen”).

Tipps
- Unsicherheiten bei Resultaten immer hinterfragen (z.B. würde sich etwas ändern wenn wir ein andere Partitionierung haben?)!
- Wir sollten eine kleine Sammlung mit Ideen erstellen, durch welche wir uns leiten lassen können.

Administratives:
- Zwischenpräsentation wird am 4 Juni 10:00 gehalten. Sie sollte ähnlich wie Abschlusspräsentation aufgebaut sein. Evtl auch Leseprobe bis dann haben.
- Stephan würde noch gerne eine Kopie vom UAP Algorithmus und dem Loss bekommen, um nachzuvollziehen, was wir gemacht haben.
